src/Expect.elm

```diff
@@ -647,3 +637,53 @@ allHelp list query =
 
                 outcome ->
                     outcome
+
+
+
+{---- Private helper functions ----}
+
+
+reportFailure : String -> String -> String -> Expectation
+reportFailure comparison expected actual =
+    Test.Expectation.Fail
+        { given = ""
+        , description = comparison
+        , reason = Test.Expectation.Comparison (toString expected) (toString actual)
+        }
+
+
+reportCollectionFailure : String -> a -> b -> List c -> List d -> Expectation
+reportCollectionFailure comparison expected actual missingKeys extraKeys =
+    Test.Expectation.Fail
+        { given = ""
+        , description = comparison
+        , reason =
+            Test.Expectation.CollectionDiff
+                { expected = toString expected
+                , actual = toString actual
+                , extra = List.map toString extraKeys
+                , missing = List.map toString missingKeys
+                }
```

Comment:
I disagree. Having the labels present makes it easier to see that the fields are being filled in correctly, and does not require coupling to the order in which the fields are defined.

I looked at putting `reason` in a let block but that results in the same depth of indenting (with elm-format).